node/core/approval-voting/src/lib.rs

```diff
@@ -636,6 +640,13 @@ impl CurrentlyCheckingSet {
 	}
 }
 
+// A list of delayed futures that gets triggered when the waiting time has expired and it is
+// time to sign the candidate.
+#[derive(Default)]
+struct SignApprovalsTimers {
+	pub timers: FuturesOrdered<BoxFuture<'static, (Hash, ValidatorIndex, CandidateHash)>>,
```

Comment:
We likely want just 1 timer per relay chain block and that should trigger signature. Otherwise we trigger signature when enough of approvals are waiting to be signed.

Reply:
I guess not in the general case, but we likely want to have most of the checks in tranche0 be coalesced and for that 1 timer is enough.

Reply:
Yes, we can achieve the same behaviour with a single timer as well.